### ISSUE TITLE
Setting Vjoy conditions '"Activate if [Button] is [released] results in a max recursion depth error

### ISSUE THREAD

**Issue by Lolo401** [muchimi/JoystickGremlinEx#12]
Hi,

First of all, a gazillion thumbs up for keeping KJG alive and adding the network control feature.  Soooo useful.

I have identified the following issue on my system : Whenever the following activation condition is used: 
     condition comparison="released" condition-type="vjoy" 
     
Then the following error message gets displayed when selecting the associated button to edit the mapping.

2024-01-20 22:55:25      ERROR Uncaught exception:
Traceback (most recent call last):
   File "gremlin\ui\common.py", line 695, in _execute_callback
   File "gremlin\ui\activation_condition.py", line 635, in _modify_vjoy
   File "gremlin\ui\activation_condition.py", line 495, in _create_ui
   File "gremlin\ui\common.py", line 758, in __init__
   File "gremlin\ui\common.py", line 559, in __init__
   File "gremlin\ui\common.py", line 762, in _initialize
   File "gremlin\joystick_handling.py", line 88, in vjoy_devices
 RecursionError: maximum recursion depth exceeded.
 
 JG Ex 13.40.3 / Vjoy 2.1.9 / HidHide 1.4.19.2.0 / Windows 11 Pro fresh install

Of note:
1. After setting the field to "released" in JG, leaving the page and coming back, the field would display "pressed".  The condition "released" never gets recorded by JG Ex. the xml profile has to be edited manually and the comparison tag changed from pressed to released. 
2. After editing the file manually, the script works as intended.

I have a significant amount of buttons relying on VJoy "released" conditions and any help would be greatly appreciated. At this time, I have to edit the xml profile with a code editor, I cannot use the JG interface due to this behavior.

Have a great day,

Lolo

**Comment by muchimi** [comment]
Thanks for the details.  I'll look at it.  I'm very busy at the moment but this should be an easy fix.

**Comment by muchimi** [comment]
Short term potential fix: 

Looks like you are using the Condition tab but that should not be necessary when using vjoyremap.  I'll have to see what happens when conditions are set (I have not touched any of that original code) but what I had in mind when implementing Vjoyremap is to use the checkbox on the main panel under the output selector that says "exec on release" which triggers the action when the source button is released (so similar to the condition tab but without the need to actually set a condition).

I suspect the infinite loop is caused by a drop-down box somewhere as I am using a newer version of QT and changes internal in QT have caused similar issues in other dialog boxes because the update logic is a bit different.

I'm not able to duplicate the "Released" radio button going to pressed - it could be the UI not updating correctly because of that condition.

Looking into it some more...

**Comment by muchimi** [comment]
One more ask - can you screenshot your action tab and the condition tab so I can duplicate it here?
thanks!

**Comment by Lolo401** [comment]
More readable:
![image](https://github.com/muchimi/JoystickGremlinEx/assets/157264378/887a5381-31ed-4ec0-bc17-9f2e6858414e)

**Comment by Lolo401** [comment]
After acknowledging the error multiple times, the following was eventually displayed: 
![image](https://github.com/muchimi/JoystickGremlinEx/assets/157264378/103734e0-4afe-41ff-baa2-532e21305689)

**Comment by Lolo401** [comment]
Hey, just wanted to confirm with you the exact installation procedure and system requirements for JG Ex.  
After a clean install of windows, I tried to install JG Ex without installing JG 13.3 first which resulted in several errors at launch.  There does not seem to be an installation process associated with JG Ex (no .msi).  Therefore, my config is based on the following: I installed 13.3 first, launched it. I am now opening JG Ex via the .exe file.  

Is that the correct way to install JG Ex ?  I did not install any other dependencies or what not.

Cheers and again, many thanks for keeping JG alive !
